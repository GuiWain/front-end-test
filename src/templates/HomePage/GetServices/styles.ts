import styled, { css } from 'styled-components'

export type ModifiersProps = {
  modifier: 'light' | 'medium' | 'semiBold' | 'bold'
}

export const Wrapper = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondaryLightest};
    width: 100%;
    padding: 3rem 0;
    margin: 2rem auto;
  `}
`

export const H4 = styled.h4`
  ${({ theme }) => css`
    color: ${theme.colors.primaryMedium};
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.semiBold};
    line-height: 4.9rem;
    letter-spacing: 0.18rem;
    text-align: center;
    margin: 2rem;
  `}
`

export const DivSteps = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  gap: 3rem;
  margin: 0 auto;
  width: 80%;
`

export const DivStep = styled.div`
  max-width: 20rem;
  height: auto;
  padding: 1.5rem;
`

export const Number = styled.p<ModifiersProps>`
  ${({ theme, modifier }) => css`
    color: ${theme.colors.secondaryMedium};
    font-size: ${theme.font.sizes.huge};
    font-weight: ${theme.font.semiBold};
    text-align: center;
    position: relative;

    display: flex;
    justify-content: center;
    margin: 0 auto;
    max-width: 50%;
    height: auto;
    z-index: 1;

    &:before {
      content: '';
      width: 8rem;
      height: 1.7rem;
      border-radius: 1rem;
      background-color: ${theme.colors.primaryMedium};
      position: absolute;
      top: 65%;
      opacity: 0.3;
      z-index: -1;
    }

    ${!!modifier && wrapperModifiers[modifier]}
  `}
`

export const Step = styled.h5`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryDarkest};
    font-size: ${theme.font.sizes.xsmall};
    text-align: center;
    margin: 0.5rem 0;
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryLight};
    font-size: ${theme.font.sizes.xxxsmall};
    line-height: 2rem;
    text-align: center;
  `}
`

const wrapperModifiers = {
  light: () => css`
    &:before {
      opacity: 0.25;
    }
  `,
  medium: () => css`
    &:before {
      opacity: 0.5;
    }
  `,
  semiBold: () => css`
    &:before {
      opacity: 0.75;
    }
  `,
  bold: () => css`
    &:before {
      opacity: 0.97;
    }
  `
}
