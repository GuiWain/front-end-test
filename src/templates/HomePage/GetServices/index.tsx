import * as S from './styles'

const GetServices = () => (
  <S.Wrapper>
    <S.H4>At lectus urna duis convallis</S.H4>
    <S.DivSteps>
      <S.DivStep>
        <S.Number modifier="light">01.</S.Number>
        <S.Step>Lorem ipsum dolor</S.Step>
        <S.Description>
          Excepteur sint occaecat cupidatat non proident.
        </S.Description>
      </S.DivStep>
      <S.DivStep>
        <S.Number modifier="medium">02.</S.Number>
        <S.Step>Cursus in hac habitasse platea dictumst.</S.Step>
        <S.Description>
          Excepteur sint occaecat cupidatat non proident.
        </S.Description>
      </S.DivStep>
      <S.DivStep>
        <S.Number modifier="semiBold">03.</S.Number>
        <S.Step>Non blandit massa</S.Step>
        <S.Description>
          Excepteur sint occaecat cupidatat non proident.
        </S.Description>
      </S.DivStep>
      <S.DivStep>
        <S.Number modifier="bold">04.</S.Number>
        <S.Step>Aliquet lectus proin nibh nisl</S.Step>
        <S.Description>
          Excepteur sint occaecat cupidatat non proident.
        </S.Description>
      </S.DivStep>
    </S.DivSteps>
  </S.Wrapper>
)

export default GetServices
