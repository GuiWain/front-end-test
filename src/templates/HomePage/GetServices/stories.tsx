import { Story, Meta } from '@storybook/react'
import GetServices from '.'

export default {
  title: 'Modules/GetServices',
  component: GetServices
} as Meta

export const Default: Story = () => <GetServices />
