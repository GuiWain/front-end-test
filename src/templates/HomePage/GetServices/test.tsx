import { render } from 'utils/test-utils'

import GetServices from '.'

describe('<GetServices />', () => {
  it('should render the GetServices', () => {
    const { container } = render(<GetServices />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
