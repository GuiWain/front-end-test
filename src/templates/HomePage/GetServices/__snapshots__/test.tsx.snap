// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetServices /> should render the GetServices 1`] = `
.c0 {
  background-color: #fafafa;
  width: 100%;
  padding: 3rem 0;
  margin: 2rem auto;
}

.c1 {
  color: #510161;
  font-size: 3.2rem;
  font-weight: 600;
  line-height: 4.9rem;
  -webkit-letter-spacing: 0.18rem;
  -moz-letter-spacing: 0.18rem;
  -ms-letter-spacing: 0.18rem;
  letter-spacing: 0.18rem;
  text-align: center;
  margin: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 3rem;
  margin: 0 auto;
  width: 80%;
}

.c3 {
  max-width: 20rem;
  height: auto;
  padding: 1.5rem;
}

.c4 {
  color: #848080;
  font-size: 4.0rem;
  font-weight: 600;
  text-align: center;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  max-width: 50%;
  height: auto;
  z-index: 1;
}

.c4:before {
  content: '';
  width: 8rem;
  height: 1.7rem;
  border-radius: 1rem;
  background-color: #510161;
  position: absolute;
  top: 65%;
  opacity: 0.3;
  z-index: -1;
}

.c4:before {
  opacity: 0.25;
}

.c7 {
  color: #848080;
  font-size: 4.0rem;
  font-weight: 600;
  text-align: center;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  max-width: 50%;
  height: auto;
  z-index: 1;
}

.c7:before {
  content: '';
  width: 8rem;
  height: 1.7rem;
  border-radius: 1rem;
  background-color: #510161;
  position: absolute;
  top: 65%;
  opacity: 0.3;
  z-index: -1;
}

.c7:before {
  opacity: 0.5;
}

.c8 {
  color: #848080;
  font-size: 4.0rem;
  font-weight: 600;
  text-align: center;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  max-width: 50%;
  height: auto;
  z-index: 1;
}

.c8:before {
  content: '';
  width: 8rem;
  height: 1.7rem;
  border-radius: 1rem;
  background-color: #510161;
  position: absolute;
  top: 65%;
  opacity: 0.3;
  z-index: -1;
}

.c8:before {
  opacity: 0.75;
}

.c9 {
  color: #848080;
  font-size: 4.0rem;
  font-weight: 600;
  text-align: center;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  max-width: 50%;
  height: auto;
  z-index: 1;
}

.c9:before {
  content: '';
  width: 8rem;
  height: 1.7rem;
  border-radius: 1rem;
  background-color: #510161;
  position: absolute;
  top: 65%;
  opacity: 0.3;
  z-index: -1;
}

.c9:before {
  opacity: 0.97;
}

.c5 {
  color: #2D2A2A;
  font-size: 1.6rem;
  text-align: center;
  margin: 0.5rem 0;
}

.c6 {
  color: #CCC8C8;
  font-size: 1.2rem;
  line-height: 2rem;
  text-align: center;
}

<section
  class="c0"
>
  <h4
    class="c1"
  >
    At lectus urna duis convallis
  </h4>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <p
        class="c4"
      >
        01.
      </p>
      <h5
        class="c5"
      >
        Lorem ipsum dolor
      </h5>
      <p
        class="c6"
      >
        Excepteur sint occaecat cupidatat non proident.
      </p>
    </div>
    <div
      class="c3"
    >
      <p
        class="c7"
      >
        02.
      </p>
      <h5
        class="c5"
      >
        Cursus in hac habitasse platea dictumst.
      </h5>
      <p
        class="c6"
      >
        Excepteur sint occaecat cupidatat non proident.
      </p>
    </div>
    <div
      class="c3"
    >
      <p
        class="c8"
      >
        03.
      </p>
      <h5
        class="c5"
      >
        Non blandit massa
      </h5>
      <p
        class="c6"
      >
        Excepteur sint occaecat cupidatat non proident.
      </p>
    </div>
    <div
      class="c3"
    >
      <p
        class="c9"
      >
        04.
      </p>
      <h5
        class="c5"
      >
        Aliquet lectus proin nibh nisl
      </h5>
      <p
        class="c6"
      >
        Excepteur sint occaecat cupidatat non proident.
      </p>
    </div>
  </div>
</section>
`;
